### PR TITLE
Improve cloud-init error messages

### LINKED ISSUE #3693: Error loading cloud-init config: bad file - multipass and cloud-init
**Describe the bug**
I get the error below when trying to create vms with multipass and cloud init.
```
error loading cloud-init config: bad file
error loading cloud-init config: bad file
error loading cloud-init config: bad file
error loading cloud-init config: bad file
```

**To Reproduce**
1. install multipass `1.14.0`
2. install cloud init `24.2-0ubuntu1~24.04.2`
3. create a cloud-init config file in config/debian-cloud-init.yaml:
```
#cloud-config
package_update: true
packages:
  - debootstrap
users:
  - name: root
    lock_passwd: false
    passwd: tersu
ssh_pwauth: true
disable_root: false
runcmd:
  - debootstrap --arch=arm64 bookworm /mnt http://deb.debian.org/debian/
  - mount --bind /dev /mnt/dev
  - mount --bind /proc /mnt/proc
  - mount --bind /sys /mnt/sys
  - |
      chroot /mnt /bin/bash -c "apt-get update && \
      apt-get install -y linux-image-arm64 systemd-sysv"
  - chroot /mnt /bin/bash -c "apt-get install -y grub-efi"
  - |
      chroot /mnt /bin/bash -c "grub-install --target=arm64-efi \
      --efi-directory=/boot/efi --bootloader-debian --recheck"
  - chroot /mnt /bin/bash -c "update-grub"
  - |
      chroot /mnt /bin/bash -c "echo 'deb http://deb.debian.org/debian/ \
      bookworm main' > /etc/apt/sources.list"
  - chroot /mnt /bin/bash -c "echo 'root:password' | chpasswd"
  - |
      chroot /mnt /bin/bash -c "sed -i 's/^#PermitRootLogin.*/\
      PermitRootLogin yes/' /etc/ssh/sshd_config"
  - chroot /mnt /bin/bash -c "systemctl enable ssh"
  - umount /mnt/dev
  - umount /mnt/proc
  - umount /mnt/sys
  - echo "Debian installation completed!"
```

4. Create a bash script to create vms using multipass and cloud-init
```
#!/bin/bash

# Launch VMs with multipass
multipass launch --name jumpbox --cpus 1 --memory 512M --disk 5G --cloud-init ./config/debian-cloud-init.yaml --verbose
multipass launch --name master --cpus 1 --memory 2G --disk 10G --cloud-init ./config/debian-cloud-init.yaml --verbose
multipass launch --name worker1 --cpus 1 --memory 2G --disk 10G --cloud-init ./config/debian-cloud-init.yaml --verbose
multipass launch --name workder2 --cpus 1 --memory 2G --disk 10G --cloud-init ./config/debian-cloud-init.yaml --verbose
```

5. Run the bash script

**Expected behavior**
It should create the multipass VMs with the specified config

**Logs**
[multipass log file.log](https://github.com/user-attachments/files/17115735/multipass.log.file.log)

**Additional info**
 - OS: OS: Ubuntu 24.04.1 LTS x86_64
 - multipass version `1.14.0`
- cloud init version `24.2-0ubuntu1~24.04.2`
- `multipass get local.driver`: qemu

**Additional context**

config is valid
```
cloud-init schema -c config/debian-cloud-init.yaml                                                                                13:31:44
Valid schema config/debian-cloud-init.yaml
```

### PR DESCRIPTION
**Pull request by ricab**
Fix #3693.

### CODE REVIEW

File: src/client/cli/cmd/launch.cpp [hunk 1]
```diff
@@ -378,10 +378,11 @@ mp::ParseCode cmd::Launch::parse_args(mp::ArgParser* parser)
 
     if (parser->isSet(cloudInitOption))
     {
+        constexpr auto err_msg_template = "Could not load cloud-init configuration from '{}': {}\n";
```
Comment (andrei-toterman) [inline src/client/cli/cmd/launch.cpp:381]: Is the '\n' at the end necessary? You are using this template in a `println` which adds its own new line, so as a result there is an extra empty line in the terminal output.

File: src/client/cli/cmd/launch.cpp [hunk 2]
```diff
@@ -396,20 +397,22 @@ mp::ParseCode cmd::Launch::parse_args(mp::ArgParser* parser)
...
+        catch (const YAML::BadFile& e)
         {
-            cerr << "error loading cloud-init config: " << e.what() << "\n";
+            auto err_detail = fmt::format("{}\n{}", e.what(), "Please ensure that Multipass can read it.");
+            fmt::println(cerr, err_msg_template, cloudInitFile, err_detail);
             return ParseCode::CommandLineError;
         }
+        catch (const YAML::Exception& e)
+        {
+            fmt::println(cerr, err_msg_template, cloudInitFile, e.what());
```
Comment (andrei-toterman) [inline src/client/cli/cmd/launch.cpp:414]: Shouldn't this also return `ParseCode::CommandLineError`? Now, if the yaml file is invalid, the error gets printed, but the launch continues.

File: src/client/cli/cmd/launch.cpp (hunk 1 above)
Comment (ricab) [reply src/client/cli/cmd/launch.cpp:381]: Ah, of course.

File: src/client/cli/cmd/launch.cpp (hunk 2 above)
Comment (ricab) [reply src/client/cli/cmd/launch.cpp:414]: Yes, duh :facepalm:

**Review by andrei-toterman — DISMISSED** [review]
Alright, LGTM now. Thanks, @ricab! Are we waiting on any other reviewer or can this be merged now?

**Comment by andrei-toterman** [comment]
Oh, huh. Now the build fails...

**Comment by ricab** [comment]
Yeah I am looking into that. I suppose it's the update to yamlcpp...

**Comment by ricab** [comment]
Hey Andrei, hope that fixes it. Anyone is free to review, of course, but we're not waiting for anyone else.

**Review by andrei-toterman — APPROVED** [review]
Alright, I tested it again and it behaves nicely. Thanks!

**Pull request merged**